graphql-protocol/src/main/java/com/linecorp/armeria/server/graphql/protocol/AbstractGraphqlService.java

```diff
@@ -190,22 +250,23 @@ private static String toStringFromJson(String name, @Nullable Object value) {
     @SuppressWarnings("unchecked")
     private static Map<String, Object> toMapFromJson(@Nullable Object maybeMap) {
         if (maybeMap == null) {
-            return ImmutableMap.of();
+            return new HashMap<>();
         }
 
         if (maybeMap instanceof Map) {
             final Map<?, ?> map = (Map<?, ?>) maybeMap;
             if (map.isEmpty()) {
-                return ImmutableMap.of();
+                return new HashMap<>();
             }
             return (Map<String, Object>) map;
```

Comment:
I needed a modifiable map to bind the file. Thinking back now, I thought it might be good to change it to return a map that can only be modified in that part.

Reply:
Question: this hasn't been handled yet right? (Asking in case I missed anything 😅 )

Reply:
I haven't made any changes yet. 🤣

Reply:
As it stands, `#toMapFromJson` sometimes creates a mutable map even though the provided factory method is an `ImmutableMap` ctor.

e.g.
```
toMapFromJson(requestMap.get("variables"), ImmutableMap::of); // this is immutable
toMapFromJson(null, ImmutableMap::of); // but this is mutable
```

What do you think of reverting this method, and just making an explicit copy outside of this method?
```
Map<String, Object> variables = toMapFromJson(requestMap.get("variables"), ImmutableMap::of);
Map<String, Object> copied = new HashMap<>(variables);
```

Reply:
good idea. Thanks for the suggestion. 👍